gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -80,8 +85,12 @@ func NewVMKeeper(
 		iavlKey:   iavlKey,
 		acck:      acck,
 		bank:      bank,
+		domain:    chainDomain,
 		maxCycles: maxCycles,
 	}
+
+	// Namespace can be either a user or crypto address.
+	vmk.reNamespace = regexp.MustCompile(`^` + chainDomain + `/(?:r|p)/([\.~_a-zA-Z0-9]+)`)
```

Comment:
```suggestion
	vmk.reNamespace = regexp.MustCompile(`^` + regexp.QuoteMeta(chainDomain) + `/(?:r|p)/([\.~_a-zA-Z0-9]+)`)
```
otherwise it will also match `gnoxland` for `chainDomain == "gno.land"`